privilege/privileges/privileges.go

```diff
@@ -167,8 +171,38 @@ func (ps *userPrivileges) ShowGrants() []string {
 // UserPrivileges implements privilege.Checker interface.
 // This is used to check privilege for the current user.
 type UserPrivileges struct {
+	// TODO: Clean up the old implementation.
 	User  string
 	privs *userPrivileges
+
+	*Handle
+}
+
+// RequestVerification implements the Checker interface.
+func (p *UserPrivileges) RequestVerification(db, table, column string, priv mysql.PrivilegeType) bool {
+	if !Enable {
+		return true
+	}
+
+	if p.User == "" {
```

Comment:
This method of `len(p.User) == 0` is more commonly used. Anyway, it's up to you.